src/contentsharecontroller/ContentShareMediaStreamBroker.ts

```diff
@@ -0,0 +1,99 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import AudioVideoController from '../audiovideocontroller/AudioVideoController';
+import DefaultDeviceController from '../devicecontroller/DefaultDeviceController';
+import Logger from '../logger/Logger';
+import MediaStreamBroker from '../mediastreambroker/MediaStreamBroker';
+
+export default class ContentShareMediaStreamBroker implements MediaStreamBroker {
+  private _mediaStream: MediaStream;
+
+  constructor(private logger: Logger) {}
+
+  get mediaStream(): MediaStream {
+    return this._mediaStream;
+  }
+
+  set mediaStream(mediaStream: MediaStream) {
+    this._mediaStream = mediaStream;
+  }
+
+  async acquireAudioInputStream(): Promise<MediaStream> {
+    if (this._mediaStream.getAudioTracks().length === 0) {
+      return DefaultDeviceController.synthesizeAudioDevice(0) as MediaStream;
+    }
+    return this._mediaStream;
+  }
+
+  async acquireVideoInputStream(): Promise<MediaStream> {
+    return this._mediaStream;
+  }
+
+  releaseMediaStream(_mediaStreamToRelease: MediaStream): void {
+    this.logger.warn('release media stream called');
+    return;
+  }
+
+  async acquireDisplayInputStream(streamConstraints: MediaStreamConstraints): Promise<MediaStream> {
+    if (
+      streamConstraints &&
+      streamConstraints.video &&
+      // @ts-ignore
+      streamConstraints.video.mandatory &&
+      // @ts-ignore
+      streamConstraints.video.mandatory.chromeMediaSource &&
+      // @ts-ignore
+      streamConstraints.video.mandatory.chromeMediaSourceId
+    ) {
+      return navigator.mediaDevices.getUserMedia(streamConstraints);
+    }
+    // @ts-ignore https://github.com/microsoft/TypeScript/issues/31821
+    return navigator.mediaDevices.getDisplayMedia(streamConstraints);
+  }
+
+  bindToAudioVideoController(_audioVideoController: AudioVideoController): void {
+    throw new Error('unsupported');
+  }
+
+  async acquireScreenCaptureDisplayInputStream(sourceId?: string): Promise<MediaStream> {
+    return this.acquireDisplayInputStream(this.screenCaptureDisplayMediaConstraints(sourceId));
+  }
+
+  private screenCaptureDisplayMediaConstraints(sourceId?: string): MediaStreamConstraints {
+    return {
+      audio: false,
+      video: {
+        ...(!sourceId && {
```

Comment:
optional: how about using the ternary operator, since there will be two shapes from one ```sourceId``` condition?

```
...(sourceId ? {
  mandatory: {
    chromeMediaSource: 'desktop',
    chromeMediaSourceId: sourceId,
    maxFrameRate: 3,
  }
} : {
  frameRate: {
    max: 3,
  }
}),
```

Reply:
This code was copied from here: https://github.com/aws/amazon-chime-sdk-js/blob/904bb29e4da9ab7ef7ea4a64298f3df25c649d76/src/screensharingsession/ScreenSharingSessionContainer.ts#L56

Let's leave it as is since we know it works and refactor later.